### PR TITLE
Configuring automation controller on OCP

### PR DESCRIPTION
**Pull request by EMcWhinn**
Adding chapter for configuring controller on OCP to minimize outage during upgrade

Document best practices and configuration requirement for tolerating k8s upgrade

https://issues.redhat.com/browse/AAP-12903

Affects `titles/ocp_performance-guide`

### CODE REVIEW

File: downstream/modules/platform/proc-configure-controller-OCP.adoc [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+During a Kubernetes upgrade, {ControllerName} has to be running. The following procedure outlines the configuration changes you need to make in automation controller to minimize downtime during the upgrade.
+
+.Prerequisites
+
+* {PlatformNameShort} 2.4
+* Ansible {ControllerName} 4.4
+* {OCPShort}
+** 4.10.42
+** 4.11.16
+** 4.12.0
```
Comment (TheRealHaoLiu) [inline downstream/modules/platform/proc-configure-controller-OCP.adoc:12]: these are minimal required versions

File: downstream/modules/platform/proc-configure-controller-OCP.adoc [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+* High availability (HA) deployment of Postgres
+* Multiple worker node that {ControllerName} pods can be scheduled on
+
+.Procedure
+
+. Enable `RECEPTOR_KUBE_SUPPORT_RECONNECT` in {ControllerName} spec.
```
Comment (TheRealHaoLiu) [inline downstream/modules/platform/proc-configure-controller-OCP.adoc:18]: `{ControllerName}`

This is specifically refering to the kubernetes resource type AutomationController

File: downstream/modules/platform/proc-configure-controller-OCP.adoc [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
++
+-----
+ee_extra_env: |
+    - name: RECEPTOR_KUBE_SUPPORT_RECONNECT
+      value: enabled
```
Comment (TheRealHaoLiu) [inline downstream/modules/platform/proc-configure-controller-OCP.adoc:23]: maybe its more clear if we do something like 

```
apiVersion: automationcontroller.ansible.com/v1beta1
kind: AutomationController
metadata:
  ...
spec:
  ...
  ee_extra_env: |
    - name: RECEPTOR_KUBE_SUPPORT_RECONNECT
      value: enabled
    ```
    
 to communicate that its suppose to be in the spec of the resource

File: downstream/modules/platform/proc-configure-controller-OCP.adoc [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+-----
++
+. Enable the graceful termination feature in {ControllerName} spec.
++
+-----
+termination_grace_period_seconds: <time to wait for job to finish>
+-----
```
Comment (TheRealHaoLiu) [inline downstream/modules/platform/proc-configure-controller-OCP.adoc:30]: same as comment above

File: downstream/modules/platform/proc-configure-controller-OCP.adoc [hunk 5]
```diff
@@ -0,0 +1,106 @@
+[id="configure-controller-OCP"]
+
+During a Kubernetes upgrade, {ControllerName} has to be running. The following procedure outlines the configuration changes you need to make in automation controller to minimize downtime during the upgrade.
```
Comment (TheRealHaoLiu) [inline downstream/modules/platform/proc-configure-controller-OCP.adoc:3]: this also help with any "voluntary disruption"

https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#voluntary-and-involuntary-disruptions

this shows more information about what voluntary disruptions are in kubernetes

File: downstream/modules/platform/proc-configure-controller-OCP.adoc [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+* {OCPShort}
+** > 4.10.42
+** > 4.11.16
+** > 4.12.0
+* High availability (HA) deployment of Postgres
+* Multiple worker node that {ControllerName} pods can be scheduled on
+
+.Procedure
+
+. Enable `RECEPTOR_KUBE_SUPPORT_RECONNECT` in AutomationController spec:
```
Comment (ianf77) [inline downstream/modules/platform/proc-configure-controller-OCP.adoc:20]: Maybe specification instead of spec?

**Review by ianf77 — COMMENTED** [review]
One comment, that can be ignored.

**Review by ianf77 — APPROVED** [review]
LGTM

**Pull request merged**